### ISSUE TITLE
Is there a way to forceUpdate/refresh the state on all global components?

### ISSUE THREAD

**Issue by mitchdowney** [CharlesStover/reactn#159]
We have a light/dark mode switching feature in our app. All of our components are changing their styles based on a theme property available on `this.global`. The different modes work if we restart our app and start in either light/dark mode.

But at the moment we switch modes, all of the components that have already rendered in the app do not update/refresh to use the proper theme. All newly rendered elements, however, use the proper theme.

Basically it appears we need to refresh every component that is already rendered in our app right after the user switches light/dark mode. In other words, it seems like we need a `this.global.forceUpdate` or something to that effect.

Is this a situation reactn can handle? Is there a method we can call to refresh global state everywhere in the app?

Thanks for the library. It has been working great for us :)

**Comment by CharlesStover** [comment]
If you have some sort of `this.global.isDarkMode` state, your components should already be rerendering when the global state changes. This sounds like you are using some additional global state (like `window.` or module state) that is tracking the global state.

There is no "refresh all components" function, and there shouldn't be. Components shouldn't be relying on global state values that do not trigger a re-render on change. The ReactN global state does trigger a re-render on change, therefore I can only conclude that you are using a different global state to determine which mode to render the component.

I would need more information to offer better suggestios.

**Comment by mitchdowney** [comment]
@CharlesStover I'm sure you're right. I'll dig deeper to figure out what's causing our issue. Thanks a lot for the info.

**Comment by mitchdowney** [comment]
Turns out I didn't understand that getGlobal doesn't listen for state updates, so I was using it interchangeably with useGlobal 🤦‍♂️ Changing getGlobal to useGlobal everywhere it was needed fixed my issue. Thanks again.

**Comment by CharlesStover** [comment]
Awesome. I'm glad you got this resolved. :)
